File: app/schemas/lead.py
from datetime import datetime

from pydantic import BaseModel, EmailStr, Field


class LeadCreate(BaseModel):
    name: str = Field(max_length=255)
    email: EmailStr
    phone: str | None = Field(default=None, max_length=50)
    company: str | None = Field(default=None, max_length=255)
    title: str | None = Field(default=None, max_length=255)
    source: str | None = Field(default=None, max_length=100)
    apollo_id: str | None = Field(default=None, max_length=100)
    status: str = Field(default="new", max_length=50)
    enrichment_data: dict | None = None


class LeadUpdate(BaseModel):
    name: str | None = Field(default=None, max_length=255)
    email: EmailStr | None = None
    phone: str | None = Field(default=None, max_length=50)
    company: str | None = Field(default=None, max_length=255)
    title: str | None = Field(default=None, max_length=255)
    source: str | None = Field(default=None, max_length=100)
    apollo_id: str | None = Field(default=None, max_length=100)
    status: str | None = Field(default=None, max_length=50)
    score: int | None = None
    enrichment_data: dict | None = None


class LeadResponse(BaseModel):
    model_config = {"from_attributes": True}

    id: int
    client_id: int
    name: str
    email: str
    phone: str | None
    company: str | None
    title: str | None
    source: str | None
    apollo_id: str | None
    status: str
    score: int | None
    enrichment_data: dict | None
    enrichment_status: str
    score_details: dict | None
    ai_analysis: dict | None
    ai_analyzed_at: datetime | None
    ai_status: str | None
    created_at: datetime
    updated_at: datetime


class LeadListResponse(BaseModel):
    items: list[LeadResponse]
    total: int
    limit: int
    offset: int


class EnrichmentLogResponse(BaseModel):
    model_config = {"from_attributes": True}

    id: int
    provider: str
    raw_response: dict | None
    success: bool
    created_at: datetime


class RoutingLogResponse(BaseModel):
    model_config = {"from_attributes": True}

    id: int
    destination: str
    payload: dict | None
    response_code: int | None
    success: bool
    error: str | None
    created_at: datetime


class LeadDetailResponse(LeadResponse):
    enrichment_logs: list[EnrichmentLogResponse]
    routing_logs: list[RoutingLogResponse]


class BulkImportRow(BaseModel):
    row: int
    errors: list[str]


class BulkImportResponse(BaseModel):
    total: int
    created: int
    updated: int
    skipped: int
    failed: int
    errors: list[BulkImportRow]

File: frontend/src/types/lead.ts
export interface Lead {
  id: number;
  client_id: number;
  name: string;
  email: string;
  phone: string | null;
  company: string | null;
  title: string | null;
  source: string | null;
  status: string;
  score: number | null;
  enrichment_data: Record<string, unknown> | null;
  enrichment_status: string;
  score_details: Record<string, unknown> | null;
  ai_analysis: AIAnalysis | null;
  ai_analyzed_at: string | null;
  ai_status: "analyzing" | "completed" | "failed" | null;
  created_at: string;
  updated_at: string;
}

export interface AIQualification {
  rating: "hot" | "warm" | "cold";
  reasoning: string;
}

export interface AIAnalysis {
  company_summary: string;
  icebreakers: string[];
  qualification: AIQualification;
  email_angle: string;
}

export interface EnrichmentLog {
  id: number;
  provider: string;
  raw_response: Record<string, unknown> | null;
  success: boolean;
  created_at: string;
}

export interface RoutingLog {
  id: number;
  destination: string;
  payload: Record<string, unknown> | null;
  response_code: number | null;
  success: boolean;
  error: string | null;
  created_at: string;
}

export interface LeadDetail extends Lead {
  enrichment_logs: EnrichmentLog[];
  routing_logs: RoutingLog[];
}

export interface LeadListResponse {
  items: Lead[];
  total: number;
  limit: number;
  offset: number;
}

// ── Export types ──────────────────────────────────────────────────────────────

export interface LeadFiltersExport {
  source?: string;
  status?: string;
  score_min?: number;
  score_max?: number;
  date_from?: string;
  date_to?: string;
  search?: string;
}

export interface WebhookExportRequest {
  webhook_url: string;
  filters?: LeadFiltersExport;
  batch_size?: number;
  include_enrichment?: boolean;
  include_ai_analysis?: boolean;
}

export interface WebhookExportResponse {
  export_id: string;
  total_leads: number;
  total_batches: number;
  status: string;
  webhook_url: string;
}
